packages/block-library/src/image/interactivity.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Internal dependencies
+ */
+import { store } from '../utils/interactivity';
+
+const raf = window.requestAnimationFrame;
+// Until useSignalEffects is fixed: https://github.com/preactjs/signals/issues/228
+const tick = () => new Promise( ( r ) => raf( () => raf( r ) ) );
+
+store( {
+	actions: {
+		core: {
+			image: {
+				showLightbox: ( { context } ) => {
+					context.core.image.initialized = true;
+					context.core.image.lightboxEnabled = true;
+					context.core.image.lastFocusedElement =
+						window.document.activeElement;
+					context.core.image.scrollPosition = window.scrollY;
+				},
+				hideLightbox: ( { context, event } ) => {
+					if ( context.core.image.lightboxEnabled ) {
+						// If scrolling, wait a moment before closing the lightbox.
+						if (
+							event.type === 'mousewheel' &&
+							Math.abs(
+								window.scrollY -
+									context.core.image.scrollPosition
+							) < 5
+						) {
+							return;
+						}
+						context.core.image.lightboxEnabled = false;
+
+						// We only want to focus the last focused element
+						// if the lightbox was closed by the keyboard.
+						// Note: Pressing Enter on a button will trigger
+						// a click event with a blank pointerType.
+						if (
+							( event.key && event.type === 'keydown' ) ||
+							( event.type === 'click' &&
+								event.pointerType === '' )
+						) {
+							context.core.image.lastFocusedElement.focus();
```

Comment:
In my testing, this is failing for the Enter and Space keys when a screen reader is running; the focus is set correctly using Esc and Tab (although tab shouldn't close the dialog.) This is probably because the screen reader is intercepting keypresses at some level. Tested with NVDA/Chrome and NVDA/Firefox. Focus is being lost.

Reply:
In which case we don't want to focus the `lastFocusedElement` when the lightbox is hidden? Couldn't we just move the `context.core.image.lastFocusedElement.focus();` outside of the conditional?

Reply:
I was just testing it and removing the conditional, and just calling `context.core.image.lastFocusedElement.focus()`,  seems to solve the issue with NVDA. It works with Enter and Space as well.

Don't we want to focus the last element when the lightbox is hidden after scrolling or when clicking outside?